src/middlewared/middlewared/plugins/nfs_/status.py

```diff
@@ -0,0 +1,32 @@
+from psutil._pslinux import Connections
+
+from middlewared.service import Service
+
+
+class NFSService(Service):
+    def client_count(self):
+        """
+        Return currently connected clients count.
+        """
+
+        """
+        As NFS does not have an explicit umount request, Ganesha does not remove clients and the following snippet
+        does not work correctly:
+
+        bus = dbus.SystemBus()
+        proxy = bus.get_object("org.ganesha.nfsd", "/org/ganesha/nfsd/ClientMgr")
+        iface = dbus.Interface(proxy, dbus_interface="org.ganesha.nfsd.clientmgr")
+        return len(iface.ShowClients()[1])
+
+        We should reconsider our approach if they ever fix this.
+        """
+
+        with open("/var/run/ganesha.pid") as f:
+            pid = int(f.read())
```

Comment:
What happens if ganesha fails to write the PID into the file?

Reply:
Is this a normal state for Ganesha? What do you think will be better approach, retrieve PID from systemd? List processes and match by cmdline?

Reply:
If no PID, casting to int will fail and be caught like you said up above so no need to complicate it now.